algebird-core/src/main/scala/com/twitter/algebird/BloomFilter.scala

```diff
@@ -20,49 +20,156 @@ import scala.collection.immutable.BitSet
 import scala.collection.JavaConverters._
 
 import com.googlecode.javaewah.{ EWAHCompressedBitmap => CBitSet }
+import com.googlecode.javaewah.IntIterator
 
 object RichCBitSet {
-  def apply(x: Int*) = {
-    CBitSet.bitmapOf(x.sorted: _*)
+  def apply(xs: Int*): CBitSet = fromArray(xs.toArray)
+
+  // this sorts the array in-place
+  def fromArray(x: Array[Int]): CBitSet = {
+    val bs = new CBitSet
+    bs += x
+  }
+  def fromBitSet(bs: BitSet): CBitSet = {
+    val nbs = new CBitSet
+    val it = bs.iterator
+    while (it.hasNext) { nbs.set(it.next) }
+    nbs
   }
   implicit def cb2rcb(cb: CBitSet): RichCBitSet = new RichCBitSet(cb)
 }
 
 // An enrichment to give some scala-like operators to the compressed
 // bit set.
-class RichCBitSet(val cb: CBitSet) {
+class RichCBitSet(val cb: CBitSet) extends AnyVal {
   def ++(b: CBitSet): CBitSet = cb.or(b)
 
   def ==(b: CBitSet): Boolean = cb.equals(b)
 
+  def +=(xs: Array[Int]): cb.type = {
+    var idx = 0
+    java.util.Arrays.sort(xs)
```

Comment:
In general, I'd say no, but since this is just a little convenience class for this code, and we only pass `Arrays` below that are ephemeral, I think it is okay.